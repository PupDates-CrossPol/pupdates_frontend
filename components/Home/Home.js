import * as React from 'react';
import { StyleSheet,  View, TextInput, Button, TouchableOpacity, Image, ScrollView } from 'react-native';
import * as ImagePicker from 'expo-image-picker';
import * as Permissions from 'expo-permissions';
import * as Camera from 'expo-camera';
import * as Constants from 'expo-constants';
import * as Font from 'expo-font';
import { Ionicons } from '@expo/vector-icons';
import { LinearGradient } from 'expo-linear-gradient';
import {createAppContainer} from 'react-navigation';
import { createStackNavigator } from 'react-navigation-stack';
import { Container, Header, Content, Card, CardItem, Text, Body } from 'native-base';

class HomeScreen extends React.Component {
  state = {
    image: null
  }

  static navigationOptions = ({navigation}) => ({
    headerLeft: () => <Image source={require('../../assets/PupDatesLogo.png')} style={styles.logo} />,
    headerTitle: () => <Image source={require('../../assets/PupDatesTitleSpread.png')} style={styles.navTitle}/>,
    headerRight: () => (
      <TouchableOpacity onPress={() => navigation.navigate('Menu')}>
        <Ionicons name="ios-menu" size={50} color='rgb(21, 112, 125)'/>
      </TouchableOpacity>
    ),
    headerStyle: {
      backgroundColor: '#fff',
    },
    headerTintColor: '#fff',
    headerTitleStyle: {
      fontWeight: 'bold',
    },
    headerLeftContainerStyle: styles.leftNavIcon,
    headerRightContainerStyle: styles.rightNavIcon,
  });

  async componentDidMount() {
   await Font.loadAsync({
      'major-mono-display': require('../../assets/fonts/MajorMonoDisplay-Regular.ttf'),
    });
  }

  render() {
    return (
      <ScrollView>
      <Container>
      <Content>
        <Card style={styles.imageCard}>
          <Text style={styles.packName}> Jordan's Pack </Text>
          <CardItem style={styles.imageCardContent}>
            <Body>
              {/* <Text> */}
                <Image source={require('../../images/rose-human1pack.jpeg')} style={styles.image} />
              {/* </Text> */}
            </Body>
          </CardItem>
          <CardItem style={styles.imageCardContent}>
            <Body style={styles.infoBody}>
              <Text style={styles.infoTextName}>Rose</Text>
              <Text style={styles.infoText}>Golden Retriever</Text>
              <Text style={styles.infoText}>3 Years Old</Text>
              <Text style={styles.infoText}>Female</Text>
              <Text style={styles.infoText}>Spayed? Yes</Text>
              <Text style={styles.infoText}>Vaccinated? Yes</Text>
              <Text style={styles.infoText}>Good with kids? Yes</Text>
            </Body>
          </CardItem>
        </Card>
      </Content>
    </Container>
    </ScrollView>

    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    marginTop: 90
  },
  logo: {
    height: 40,
    width: 40,
},
  // image: {
  //   height: 140,
  //   width: 140,
  // },
  title: {
    fontSize: 50,
  },
  imageCard: {
    flexDirection: 'column',
    alignItems: 'center',
    justifyContent: 'center',
    marginTop: 15,
  },
  infoCard: {
    flexDirection: 'column',
    alignItems: 'center',
    justifyContent: 'center',
    marginTop: 10,
    width: '80%'
  },
  packName: {
    fontSize: 25
  },
  imageCardContent: {
    alignItems: 'center',
    justifyContent: 'center',
    flexDirection: 'column',
    height: 350,
    
  },
  buttonText: {
    color: '#fff',
    fontSize: 25,
  },
  navTitle: {
    width: 160,
    height: 40,
    marginBottom: 5,
},
rightNavIcon: { 
    marginRight: 10,  
},
leftNavIcon: { 
  marginLeft: 10,  
  marginBottom: 5, 
},
  image: {
    height: 325,
    width: 275,
    alignItems: 'center',
    justifyContent: 'center',
    borderColor: 'black',
    borderWidth: 3.5,
    borderRadius: 50
  },
  infoText: {
    textAlign: 'left',
    fontSize: 20,
    color: 'grey',
    alignSelf: 'stretch'
  },
  infoTextName: {
    fontSize: 30
  },
  infoBody: {
    alignItems: 'center',
    flexDirection: 'column',
    justifyContent: 'center',
    textAlign: 'left'
  }
});


const AppNavigator = createStackNavigator({
  Home: {
    screen: HomeScreen,
  },
});



export default createAppContainer(AppNavigator)