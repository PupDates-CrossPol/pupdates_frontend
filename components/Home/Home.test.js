import React from 'react'
import renderer from 'react-test-renderer'
import { HomeScreen as Home, mapStateToProps, mapDispatchToProps } from './Home'

describe('<Home />', () => {
  it.skip('has 2 children', () => {
    const naviProp = { navigation: { navigate: () => {} } };
    const mockUser = {
      attributes: {
        id: 1,
        first_name: 'bob',
        last_name: 'ross',
        description: 'happy little trees',
        email: 'happylittletrees@email.com',
        image: 'an image url'
      }
    }

    const mockSwipeUser = {
      attributes: {
        id: 2,
        first_name: 'sara',
        last_name: 'karsh',
        description: 'likes dogs',
        email: 'sarasemail',
      }
    }

    const mockSwipePackPhotos = [
      {
        id: 1,
        dog_id: 2,
        image_url: 'some string'
      }
    ]

    const mockSwipePack = [
      {
        attributes: 
          {
            id: 2,
            user_id: 2,
            name: 'fido',
            breed: 'dog'
          }
      }
    ]

    const tree = renderer.create(<Home navigation={naviProp} user={mockUser} swipeUser={mockSwipeUser} swipePackPhotos={mockSwipePackPhotos} swipePack={mockSwipePack} />).toJSON()
    // console.log('TREE', tree)
    expect(tree.children.length).toBe(2)
  })

  it.skip('renders correctly', () => {
    const naviProp = { navigation: { navigate: () => {} } };
    const mockUser = {
      attributes: {
        id: 1,
        first_name: 'bob',
        last_name: 'ross',
        description: 'happy little trees',
        email: 'happylittletrees@email.com',
        image: 'an image url'
      }
    }

    const mockSwipeUser = {
      attributes: {
        id: 2,
        first_name: 'sara',
        last_name: 'karsh',
        description: 'likes dogs',
        email: 'sarasemail',
      }
    }

    const mockSwipePackPhotos = [
      {
        id: 1,
        dog_id: 2,
        image_url: 'some string'
      }
    ]

    const mockSwipePack = [
      {
        id: 2,
        user_id: 2,
        name: 'fido',
        breed: 'dog'
      }
    ]

    const tree = renderer.create(<Home navigation={naviProp} user={mockUser} swipeUser={mockSwipeUser} swipePackPhotos={mockSwipePackPhotos} swipePack={mockSwipePack} />).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
