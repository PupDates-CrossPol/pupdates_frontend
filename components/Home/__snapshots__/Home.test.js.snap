// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
  <RCTScrollView>
    <View>
      <View
        style={
          Object {
            "backgroundColor": "#fff",
            "flex": 1,
            "height": 1334,
          }
        }
      >
        <RCTScrollView
          automaticallyAdjustContentInsets={false}
          contentContainerStyle={
            Array [
              Object {
                "padding": undefined,
              },
              undefined,
            ]
          }
          contentInset={
            Object {
              "bottom": 0,
            }
          }
          enableAutomaticScroll={true}
          enableOnAndroid={false}
          enableResetScrollToCoords={true}
          extraHeight={75}
          extraScrollHeight={0}
          getScrollResponder={[Function]}
          handleOnScroll={[Function]}
          keyboardDismissMode="interactive"
          keyboardOpeningTime={250}
          keyboardShouldPersistTaps="handled"
          keyboardSpace={0}
          onScroll={[Function]}
          resetKeyboardSpace={[Function]}
          resetScrollToCoords={
            Object {
              "x": 0,
              "y": 0,
            }
          }
          scrollEventThrottle={1}
          scrollForExtraHeightOnAndroid={[Function]}
          scrollIntoView={[Function]}
          scrollToEnd={[Function]}
          scrollToFocusedInput={[Function]}
          scrollToPosition={[Function]}
          showsVerticalScrollIndicator={true}
          style={
            Object {
              "backgroundColor": "transparent",
              "flex": 1,
            }
          }
          update={[Function]}
          viewIsInsideTabBar={false}
        >
          <View>
            <Text
              style={
                Array [
                  Object {
                    "color": "#000",
                    "fontFamily": "System",
                    "fontSize": 16,
                  },
                  Object {
                    "fontSize": 25,
                    "textAlign": "center",
                  },
                ]
              }
              uppercase={false}
            >
              sara
              's Pack
            </Text>
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "#fff",
                    "borderBottomWidth": 0.5,
                    "borderColor": "#ccc",
                    "borderLeftWidth": 0.5,
                    "borderRadius": 2,
                    "borderRightWidth": 0.5,
                    "borderTopWidth": 0.5,
                    "elevation": 3,
                    "flexWrap": "nowrap",
                    "marginBottom": 5,
                    "marginLeft": 2,
                    "marginRight": 2,
                    "marginTop": 5,
                    "shadowColor": "#000",
                    "shadowOffset": Object {
                      "height": 2,
                      "width": 0,
                    },
                    "shadowOpacity": 0.1,
                    "shadowRadius": 1.5,
                  },
                  Object {
                    "alignItems": "center",
                    "flexDirection": "column",
                    "justifyContent": "center",
                    "marginTop": 15,
                  },
                ]
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#fff",
                      "borderRadius": 2,
                      "flexDirection": "row",
                      "paddingBottom": 10,
                      "paddingLeft": 15,
                      "paddingRight": 15,
                      "paddingTop": 10,
                    },
                    Object {
                      "alignItems": "center",
                      "flexDirection": "column",
                      "height": 350,
                      "justifyContent": "center",
                    },
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "flex-start",
                      "alignSelf": "stretch",
                      "flex": 1,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "borderRadius": 2,
                      }
                    }
                  >
                    <RCTScrollView
                      ImageComponent={[Function]}
                      activeAnimationOptions={null}
                      activeAnimationType="timing"
                      activeSlideAlignment="center"
                      activeSlideOffset={20}
                      apparitionDelay={0}
                      automaticallyAdjustContentInsets={false}
                      autoplay={false}
                      autoplayDelay={1000}
                      autoplayInterval={3000}
                      borderTopLeftRadius={2}
                      borderTopRightRadius={2}
                      callbackOffsetMargin={5}
                      circleLoop={true}
                      containerCustomStyle={Object {}}
                      contentContainerCustomStyle={Object {}}
                      contentContainerStyle={
                        Array [
                          Object {
                            "paddingLeft": 0,
                            "paddingRight": 0,
                          },
                          Object {},
                        ]
                      }
                      data={
                        Array [
                          "some string",
                        ]
                      }
                      decelerationRate="fast"
                      directionalLockEnabled={true}
                      disableVirtualization={false}
                      dotColor="rgb(21, 112, 125)"
                      enableMomentum={false}
                      enableSnap={true}
                      firstItem={0}
                      getItem={[Function]}
                      getItemCount={[Function]}
                      hasParallaxImages={false}
                      horizontal={true}
                      images={
                        Array [
                          "some string",
                        ]
                      }
                      inactiveSlideOpacity={0.7}
                      inactiveSlideScale={0.9}
                      inactiveSlideShift={0}
                      initialNumToRender={6}
                      invertStickyHeaders={false}
                      inverted={false}
                      itemWidth={750}
                      keyExtractor={[Function]}
                      layout="default"
                      lockScrollTimeoutDuration={1000}
                      lockScrollWhileSnapping={false}
                      loop={true}
                      loopClonesPerSide={5}
                      maxToRenderPerBatch={13}
                      numColumns={1}
                      onContentSizeChange={[Function]}
                      onEndReachedThreshold={2}
                      onLayout={[Function]}
                      onMomentumScrollEnd={[Function]}
                      onResponderRelease={[Function]}
                      onScroll={[Function]}
                      onScrollBeginDrag={[Function]}
                      onScrollEndDrag={[Function]}
                      onSnapToItem={[Function]}
                      onStartShouldSetResponderCapture={[Function]}
                      onTouchEnd={[Function]}
                      onTouchStart={[Function]}
                      overScrollMode="never"
                      pinchGestureEnabled={false}
                      removeClippedSubviews={true}
                      renderItem={[Function]}
                      scrollEnabled={true}
                      scrollEventThrottle={1}
                      scrollsToTop={false}
                      shouldOptimizeUpdates={true}
                      showsHorizontalScrollIndicator={false}
                      showsVerticalScrollIndicator={false}
                      slideStyle={Object {}}
                      sliderWidth={750}
                      stickyHeaderIndices={Array []}
                      style={
                        Object {
                          "flexDirection": "row",
                          "opacity": 0,
                          "width": 750,
                        }
                      }
                      swipeThreshold={20}
                      updateCellsBatchingPeriod={50}
                      useScrollView={false}
                      vertical={false}
                      viewabilityConfigCallbackPairs={Array []}
                      windowSize={13}
                    >
                      <View>
                        <View
                          onLayout={[Function]}
                          style={
                            Array [
                              Object {
                                "flexDirection": "row",
                              },
                              null,
                            ]
                          }
                        >
                          <View
                            pointerEvents="box-none"
                            style={
                              Object {
                                "opacity": 1,
                                "transform": Array [
                                  Object {
                                    "scale": 1,
                                  },
                                ],
                                "width": 750,
                              }
                            }
                          >
                            <View
                              accessible={true}
                              focusable={true}
                              isTVSelectable={true}
                              onClick={[Function]}
                              onResponderGrant={[Function]}
                              onResponderMove={[Function]}
                              onResponderRelease={[Function]}
                              onResponderTerminate={[Function]}
                              onResponderTerminationRequest={[Function]}
                              onStartShouldSetResponder={[Function]}
                              style={null}
                            >
                              <Image
                                ImageComponent={[Function]}
                                circleLoop={true}
                                dotColor="rgb(21, 112, 125)"
                                images={
                                  Array [
                                    "some string",
                                  ]
                                }
                                resizeMethod="resize"
                                resizeMode="cover"
                                source={
                                  Object {
                                    "uri": "some string",
                                  }
                                }
                                style={
                                  Object {
                                    "alignItems": "center",
                                    "alignSelf": "center",
                                    "borderColor": "black",
                                    "borderRadius": 50,
                                    "borderWidth": 3.5,
                                    "height": 325,
                                    "justifyContent": "center",
                                    "width": 275,
                                  }
                                }
                              />
                            </View>
                          </View>
                        </View>
                      </View>
                    </RCTScrollView>
                  </View>
                </View>
              </View>
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#fff",
                      "borderRadius": 2,
                      "flexDirection": "row",
                      "paddingBottom": 10,
                      "paddingLeft": 15,
                      "paddingRight": 15,
                      "paddingTop": 10,
                    },
                    Object {
                      "alignItems": "center",
                      "flexDirection": "column",
                      "height": 250,
                      "justifyContent": "center",
                      "width": "80%",
                    },
                  ]
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "flex-start",
                        "alignSelf": "stretch",
                        "flex": 1,
                      },
                      Object {
                        "flexDirection": "column",
                        "height": 50,
                        "justifyContent": "center",
                        "textAlign": "left",
                      },
                    ]
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#000",
                          "fontFamily": "System",
                          "fontSize": 16,
                        },
                        Object {
                          "fontSize": 30,
                          "textAlign": "left",
                        },
                      ]
                    }
                    uppercase={false}
                  >
                    fido
                  </Text>
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#000",
                          "fontFamily": "System",
                          "fontSize": 16,
                        },
                        Object {
                          "alignSelf": "stretch",
                          "color": "grey",
                          "fontSize": 20,
                          "textAlign": "left",
                        },
                      ]
                    }
                    uppercase={false}
                  >
                    Breed: 
                    dog
                  </Text>
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#000",
                          "fontFamily": "System",
                          "fontSize": 16,
                        },
                        Object {
                          "alignSelf": "stretch",
                          "color": "grey",
                          "fontSize": 20,
                          "textAlign": "left",
                        },
                      ]
                    }
                    uppercase={false}
                  >
                    Age: 
                  </Text>
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#000",
                          "fontFamily": "System",
                          "fontSize": 16,
                        },
                        Object {
                          "alignSelf": "stretch",
                          "color": "grey",
                          "fontSize": 20,
                          "textAlign": "left",
                        },
                      ]
                    }
                    uppercase={false}
                  >
                    Gender: 
                  </Text>
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#000",
                          "fontFamily": "System",
                          "fontSize": 16,
                        },
                        Object {
                          "alignSelf": "stretch",
                          "color": "grey",
                          "fontSize": 20,
                          "textAlign": "left",
                        },
                      ]
                    }
                    uppercase={false}
                  >
                    Fixed: 
                    false
                  </Text>
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#000",
                          "fontFamily": "System",
                          "fontSize": 16,
                        },
                        Object {
                          "alignSelf": "stretch",
                          "color": "grey",
                          "fontSize": 20,
                          "textAlign": "left",
                        },
                      ]
                    }
                    uppercase={false}
                  >
                    Vaccinated: 
                    false
                  </Text>
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#000",
                          "fontFamily": "System",
                          "fontSize": 16,
                        },
                        Object {
                          "alignSelf": "stretch",
                          "color": "grey",
                          "fontSize": 20,
                          "textAlign": "left",
                        },
                      ]
                    }
                    uppercase={false}
                  >
                    Good with kids: 
                    false
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </RCTScrollView>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "alignSelf": "center",
        "bottom": 10,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "position": "absolute",
        "width": "90%",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text />
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text />
    </View>
  </View>
</RCTSafeAreaView>
`;
